backends/lakeroad/lakeroad.cc

```diff
@@ -1714,11 +1714,23 @@ struct BtorBackend : public Backend {
 	void execute(std::ostream *&f, std::string filename, std::vector<std::string> args, RTLIL::Design *design) override
 	{
 		log_header(design, "Executing Lakeroad egglog backend.\n");
-
 		RTLIL::Module *topmod = design->top_module();
 
+		// Copied from firrtl code. Not sure why "filename" is not set ever
+		// even when I pass "write_lakeroad hello.egg" or something
```

Comment:
```suggestion
		// Copied from firrtl code.
```

I'll explain to you why filename is not set, and then we can delete this comment (b/c the comment isn't actually relevant to the code). the `filename` variable isn't set because, when you call `write_lakeroad` from the Yosys shell (which is how you've been calling it), the Yosys internals call `execute` without passing `filename`. I don't know _where_ in the Yosys codebase `execute` is ever called with a filename, but there must be some place where it's used. When you do `write_lakeroad hello.egg`, Yosys puts `hello.egg` into `args`, not into `filename`. (that's why this block of code exists: it basically says, if `filename` isn't set explicitly, then use the last argument as a filename)